pkg/clusterrouter/proxy/proxy_server_test.go

```diff
@@ -0,0 +1,165 @@
+package proxy
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	auth "github.com/rancher/rancher/pkg/auth/context"
+	"github.com/stretchr/testify/assert"
+	"io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+func TestServeHTTP(t *testing.T) {
+	tests := map[string]struct {
+		tokenCreator                   tokenCreator
+		impersonatorAccountTokenGetter impersonatorAccountTokenGetter
+		header                         http.Header
+		ctx                            func() context.Context
+		wantHeader                     http.Header
+		wantErr                        string
+	}{
+		"get impersonation token": {
+			header: map[string][]string{},
+			ctx: func() context.Context {
+				ctx := context.TODO()
+				return request.WithUser(ctx, &user.DefaultInfo{
```

Comment:
nit: This can probably be inlined here and in other test cases
```suggestion
				return request.WithUser(context.TODO(), &user.DefaultInfo{
```